daemon/access_test.go

```diff
@@ -223,3 +226,163 @@ func (s *accessSuite) TestSnapAccess(c *C) {
 	c.Check(ac.CheckAccess(nil, nil, ucred, nil), DeepEquals, errForbidden)
 	c.Check(ac.CheckAccess(nil, nil, nil, nil), DeepEquals, errForbidden)
 }
+
+func (s *accessSuite) TestRequireThemeApiAccessImpl(c *C) {
+	d := s.daemon(c)
+	s.mockSnap(c, `
+name: core
+type: os
+version: 1
+slots:
+  snap-themes-control:
+`)
+	s.mockSnap(c, `
+name: some-snap
+version: 1
+plugs:
+  snap-themes-control:
+`)
+
+	restore := daemon.MockCgroupSnapNameFromPid(func(pid int) (string, error) {
+		if pid == 42 {
+			return "some-snap", nil
+		}
+		return "", fmt.Errorf("not a snap")
+	})
+	defer restore()
+
+	var ac daemon.AccessChecker = daemon.ThemesOpenAccess{}
+
+	// Access with no ucred data is forbidden
+	c.Check(ac.CheckAccess(d, nil, nil, nil), DeepEquals, errForbidden)
+
+	// Access from snapd.socket is allowed
+	ucred := &daemon.Ucrednet{Uid: 1000, Pid: 1001, Socket: dirs.SnapdSocket}
+	c.Check(ac.CheckAccess(d, nil, ucred, nil), IsNil)
+
+	// Access from unknown sockets is forbidden
+	ucred = &daemon.Ucrednet{Uid: 1000, Pid: 1001, Socket: "unknown.socket"}
+	c.Check(ac.CheckAccess(d, nil, ucred, nil), DeepEquals, errForbidden)
+
+	// Access from snapd-snap.socket is rejected by default
+	ucred = &daemon.Ucrednet{Uid: 1000, Pid: 42, Socket: dirs.SnapSocket}
+	c.Check(ac.CheckAccess(d, nil, ucred, nil), DeepEquals, errForbidden)
+
+	// Now connect the marker interface
+	st := d.Overlord().State()
+	st.Lock()
+	st.Set("conns", map[string]interface{}{
+		"some-snap:snap-themes-control core:snap-themes-control": map[string]interface{}{
+			"interface": "snap-themes-control",
+		},
+	})
```

Comment:
We should also have a check where we've snap-themes-control interface in "conns" but with undesired flag set.

Also, it would be good to have another interface on some-snap that is connected from the very start of the test to show that it's irrelevant and only snap-themes-control counts when connected.